### PR TITLE
Add test for removing a replica and recreating it

### PR DESCRIPTION
**Pull request by keithalucas**
https://github.com/longhorn/longhorn/issues/1628

Signed-off-by: Keith Lucas <keith.lucas@suse.com>

### CODE REVIEW

File: integration/core/test_cli.py [hunk 1]
```diff
@@ -1397,3 +1400,50 @@ def test_engine_restart_after_sigkill(bin):  # NOQA
...
+    r2_url = grpc_replica_client2.url
+    v = grpc_controller_client.volume_start(replicas=[
+        r1_url, r2_url,
+    ])
+    assert v.replicaCount == 2
+
+    # Remove the first replica and recreate it immediately.
+    grpc_controller_client.replica_delete(r1_url)
+
+    for i in range(RETRY_COUNTS):
```
Comment (shuo-wu) [inline integration/core/test_cli.py:1437]: Add wait interval for this retry loop. Then the test doesn't need to retry for 100 times.

**Review by shuo-wu — CHANGES_REQUESTED** [review]
It's better to add the test skeleton before implementation.

File: integration/core/test_cli.py (hunk 1 above)
Comment (keithalucas) [reply integration/core/test_cli.py]: I added the sleep and changed it to use RETRY_COUNTS_SHORT.

**Comment by keithalucas** [comment]
> It's better to add the test skeleton before implementation.

Added the sketelon.

**Review by shuo-wu — APPROVED** [review]

**Review by joshimoo — APPROVED** [review]
LGTM

**Pull request merged**